fem/tmop.cpp

```diff
@@ -27,20 +27,20 @@ typedef internal::dual<real_t, real_t> ADFType;
 typedef internal::dual<ADFType, ADFType> ADSType;
 
 template <typename type>
-auto fnorm2_2D(std::vector<type>& u) -> type
+auto fnorm2_2D(std::vector<type> &u) -> type
```

Comment:
Are the return deduction guides necessary?

Reply:
To the best of my knowledge it is equivalent in to `type fnorm2_2D(std::vector<type> &u)` and in this case a personal preference.

Reply:
I also think that
`type ...`
is more straightforward than
`auto ... -> type`